app/src/org/commcare/activities/CommCareActivity.java

```diff
@@ -944,4 +944,8 @@ protected void showToast(int stringResource) {
     protected String getLocalizedString(int stringResource) {
         return StringUtils.getStringRobust(this, stringResource);
     }
+
+    public CommCareTask getCurrentTask() {
+        return stateHolder.getCurrentTask();

```

Comment:
One small thing - is it possible to pass this through a Protected method or otherwise not make it public to parts of the code outside of the test environment (which I think has its own base class that can reveal it)? 

It's not worth going through a lot of effort to prevent, but exposing internal representations can cause headaches down the road.

Reply:
Made some changes in https://github.com/dimagi/commcare-android/pull/2537/commits/24a58e41daefe85d549461198da981c7142afbb0. A few of the modifiers are still public for only testing and I could not figure out an easy way to do otherwise. So I have marked such methods with `@VisibleForTesting` instead.